client/src/routes.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { Switch, Route } from "react-router-dom";
+import HomePage from "./containers/HomePage";
+import AddressPage from "./containers/AddressPage";
+import BBLPage from "./containers/BBLPage";
+import AboutPage from "./containers/AboutPage";
+import HowToUsePage from "./containers/HowToUsePage";
+import MethodologyPage from "./containers/Methodology";
+import TermsOfUsePage from "./containers/TermsOfUsePage";
+import PrivacyPolicyPage from "./containers/PrivacyPolicyPage";
+
+export const createRouteForAddressPage = (boro: string, streetname: string, housenumber?: string) =>
+  `/address/${boro}/${housenumber || " "}/${streetname}`;
+
+const addressPageRouteWithParams = createRouteForAddressPage(
+  ":boro",
+  ":streetname",
+  ":housenumber"
+);
```

Comment:
Err, using `encodeURIComponent()` will break this though, as it'd escape the `:`; you'd want to just replace this with the raw string `/address/:boro/:streetname/:housenumber` rather than calling out to `createRouteForAddressPage()`.

Honestly though, IMO the street name and house number (possibly also the boro) should never have been path components, but rather querystring arguments.  Path components should only ever contain strings that don't need any escaping, like slugs, numbers, etc.  But I guess we are stuck with what we've got since we don't want to break existing links...

Reply:
Sure thing